utils/pangeaAuthCheck.ts

```diff
@@ -0,0 +1,104 @@
+import { PangeaConfig, AuthNService, PangeaErrors } from "pangea-node-sdk";
+
+const config = new PangeaConfig({ domain: process.env.PANGEA_DOMAIN });
+const authn = new AuthNService(process.env.PANGEA_TOKEN, config);
+
+const getBearerToken = (req) => {
+  const authorizationHeader =
+  req.headers instanceof Headers
+  ? req.headers.get("authorization")
+  : req.headers?.authorization;
+  
+  const authorizationHeaderParts = authorizationHeader?.split(" ");
+  
+  const bearerToken =
+  authorizationHeaderParts?.[0]?.toLowerCase() === "bearer" &&
+  authorizationHeaderParts?.[1];
+  
+  return bearerToken;
+};
+
+const validateToken = async (token: string) => {
+  const result = false;
+
+  if (token) {
```

Comment:
Instead of having this whole thing under a `if (token) {`, just check for that first and return early. Saves on indentation.

```ts
const validateToken = async (token: string) => {
  if (!token) {
    return false;
  }

  try {
    // ...
```